### ISSUE TITLE
Control classes leaking memory

### ISSUE THREAD

**Issue by dkazennov** [dotnet/winforms#7329]
### .NET version

.NET 7.0 Preview 4

### Did it work in .NET Framework?

No

### Issue description

Investigating possible memory leaks with Windows Forms classes inherited to Control class AND in corresponding Accessible Objects.
Some control items and some accessible objects are left in memory after the control instance with its items was inspected with accessibility tool (like Accessibility Insights or Inspect) and the form with the control is closed, garbage is collected, but accessibility tool is still opened.

Abstract, private and internal control classes which can't be placed in a form Controls collection are not applicable.
Control classes from the namespace System.Windows.Forms.Design are not not applicable.

**Table 1. Control classes tested for memory leaks:**

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/V-DKAZ~1.EUR/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/V-DKAZ~1.EUR/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
</head>

<body link="#0563C1" vlink="#954F72">

Class   name | Base class name | Memory leaks | Issue assigned | Status
-- | -- | -- | -- | --
DataGridView | Control | Is leaking | #7347 | Fixed
PropertyGrid | ContainerControl | Is leaking | #6943 | Fixed
DomainUpDown | UpDownBase | Is leaking | #7328 | Fixed
ToolStrip | ScrollableControl | Is leaking | #7375 | Fixed
ListBox | ListControl | Is leaking | #7197 | Fixed
ListView | Control | Is leaking | #7385 | Fixed
CheckedListBox | ListBox | Is leaking | #7197 | Fixed
ComboBox | ListControl | Is leaking | #7318 | Fixed
TextBox | TextBoxBase | Is leaking | #7248 | Work in progress
TreeView | Control | Is leaking | #7196 | Fixed
MonthCalendar | Control | Is leaking | #6852 | Fixed
NumericUpDown | UpDownBase | Is leaking | #7328 | Fixed
TabPage | Panel | Is leaking (when   tested with Narrator) | #7415 | Fixed
LinkLabel | Label | Is leaking |  #9327 | To do
Button | ButtonBase | Is not leaking |  
CheckBox | ButtonBase | Is not leaking |  
RadioButton | ButtonBase | Is not leaking |  
DateTimePicker | Control | Is not leaking |  
GroupBox | Control | Is not leaking |  
ThreadExceptionDialog | Form | Is not leaking |  
UpDownBase | ContainerControl | Is not leaking |  
SplitterPanel | Panel | Is not leaking |  
BindingNavigator | ToolStrip | Is not leaking |
MenuStrip | ToolStrip | Is not leaking |  
StatusStrip | ToolStrip | Is not leaking |  
ContextMenuStrip | ToolStripDropDownMenu | Is not leaking |  
ToolStripOverflow | ToolStripDropDown | Is not leaking |  
HScrollBar | ScrollBar | Is not leaking |  
VScrollBar | ScrollBar | Is not leaking |  
Label | Control | Is not leaking |  
ProgressBar | Control | Is not leaking |  
RichTextBox | TextBoxBase | Is not leaking | 
MaskedTextBox | TextBoxBase | Is not leaking |  
TabControl | Control | Is not leaking |  
TrackBar | Control | Is not leaking |  
DataGridViewTextBoxEditingControl | TextBox | Is not leaking |  
PrintPreviewControl | Control | Is not leaking |  
ToolStripPanel | ContainerControl | Is not leaking |  
SplitContainer | ContainerControl | Is not leaking |  
ToolStripContainer | ContainerControl | Is not leaking |  
Panel | ScrollableControl | Is not leaking |  
TableLayoutPanel | Panel | Is not leaking |  
ToolStripContentPanel | Panel | Is not leaking |  
DataGridViewComboBoxEditingControl | ComboBox | Is not leaking |  
PictureBox | Control | Is not leaking |  
ColorEditorTabControl | TabControl | Is not leaking | 
Splitter | Control | Is not leaking | 
WebBrowser | WebBrowserBase | Is not leaking | | Deprecated 

</body>

</html>

**Table 2. Control classes that are not applicable on the form:**

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/V-DKAZ~1.EUR/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/V-DKAZ~1.EUR/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
</head>

<body link="#0563C1" vlink="#954F72">

Class   name | Base class name
-- | --
Control |  
AdornerWindow | Control
AnchorUI | Control
AxHost | Control
ButtonBase | Control
DetailsButton | Button
DropDownButton | Button
EditorButton | Button
SplitButton | Button
DockEditorCheckBox | CheckBox
ColorPalette | Control
ColorUI | Control
ContainerPlaceholder | Control
ContentUI | Control
ControlPlaceholder | Control
DateTimeUI | Control
DesignerFrame | Control
DockUI | Control
GridToolTip | Control
SelectionUIService | Control
ScrollableControl | Control
ComponentTray | ScrollableControl
ContainerControl | ScrollableControl
DesignerActionPanel | ContainerControl
BinaryUI | Form
CollectionForm | Form
CollectionEditorCollectionForm | CollectionForm
StringCollectionForm | CollectionForm
ComponentEditorForm | Form
DataGridViewCellStyleBuilder | Form
DropDownHolder | Form
FormatStringDialog | Form
FlyoutDialog | Form
DropDownHolder | FlyoutDialog
GridErrorDialog | Form
PrintPreviewDialog | Form
StatusDialog | Form
LinkAreaUI | Form
MaskDesignerDialog | Form
MdiWindowDialog | Form
ParkingWindow | ContainerControl
ToolStripNumericUpDownControl | NumericUpDown
UserControl | ContainerControl
BorderSidesEditorUI | UserControl
FormatControl | UserControl
ShortcutKeysUI | UserControl
MaskedTextBoxTextEditorDropDown | UserControl
OverlayControl | ScrollableControl
ComponentEditorPage | Panel
DataGridViewEditingPanel | Panel
FlowLayoutPanel | Panel
ByteViewer | TableLayoutPanel
ToolStripEditorControl | Panel
PropertyGridToolStrip | ToolStrip
MdiControlStrip | MenuStrip
MdiWindowListStrip | MenuStrip
ToolStripDropDown | ToolStrip
DesignerActionToolStripDropDown | ToolStripDropDown
FeedbackDropDown | ToolStripDropDown
ToolStripDropDownMenu | ToolStripDropDown
GroupedContextMenuStrip | ContextMenuStrip
BaseContextMenuStrip | GroupedContextMenuStrip
NewItemsContextMenuStrip | GroupedContextMenuStrip
TransparentToolStrip | ToolStrip
ScrollBar | Control
SKWindow | Control
SpringControl | Control
EditorLabel | Label
MethodItemLinkLabel | LinkLabel
StickyLabel | Label
ListControl | Control
ColorEditorListBox | ListBox
CursorUI | ListBox
GridViewListBox | ListBox
FilterListBox | ListBox
ToolStripComboBoxControl | ComboBox
MarshalingControl | Control
MdiClient | Control
DateTimeMonthCalendar | MonthCalendar
RightToLeftLayoutGrip | Control
ToolStripProgressBarControl | ProgressBar
PropertyGridView | Control
SnappableControl | Control
CommandsPane | PropertyGrid.SnappableControl
HelpPane | PropertyGrid.SnappableControl
TabOrder | Control
TextBoxBase | Control
MultilineStringEditorUI | RichTextBox
ridViewTextBox | TextBox
mplateTextBox | TextBox
ToolStripTextBoxControl | TextBox
UpDownEdit | TextBox
ToolStripAdornerWindow | Control
TrayControl | Control
PageSelector | TreeView
Selector | TreeView
UpDownButtons | Control
WebBrowserBase | Control

</body>

</html>

### Steps to reproduce

1) Include a MemoryLeaks_Form to WinformsControlsTest (impemented in https://github.com/dkazennov/winforms/tree/MemoryLeaks_Form).
2) Test each form with WinDbg and Inspect.

**The separate application to test memory leaks** (it uses checkbox instead of buttons to switch between the forms to isolate the Button tested class memory objects):
[MemoryLeaksApp.zip](https://github.com/dotnet/winforms/files/9040731/MemoryLeaksApp.zip)

**Comment by RussKie** [comment]
To confirm - are _controls_ leaking memory, or their _AO_ implementations?

**Comment by dkazennov** [comment]
> To confirm - are _controls_ leaking memory, or their _AO_ implementations?

There are problems both with Controls and Accessible objects. This issue is to investigate the whole problem and to create subsequent issues.

**Comment by dkazennov** [comment]
Button control: 
![image](https://user-images.githubusercontent.com/102961955/175299490-7b4ebf8c-da14-4182-9b99-96af6fd08144.png)
CheckBox control: 
![image](https://user-images.githubusercontent.com/102961955/175299547-a1b4c914-b175-4b78-bb78-c438563dd448.png)
DataGridView control: 
![image](https://user-images.githubusercontent.com/102961955/175299622-3940b441-bcb7-481f-a134-531e2ee7459b.png)
DateTimePicker control:
![image](https://user-images.githubusercontent.com/102961955/175299661-4b3e1da5-c5fc-4108-ba43-cd3eafc59cb5.png)
GroupBox control:
![image](https://user-images.githubusercontent.com/102961955/175299732-eca04056-97ca-4590-a813-f758768b35aa.png)
RadioButton control:
![image](https://user-images.githubusercontent.com/102961955/175299787-b79de204-60e4-4ae1-a3b4-b7b1a01ec17a.png)

**Comment by vladimir-krestov** [comment]
ToolStrip:
![image](https://user-images.githubusercontent.com/49272759/176413197-a0981734-225b-43f1-83c0-07ce3ec6048a.png)

**Comment by dkazennov** [comment]
TabPage leaks when tested with Narrator:
![image](https://user-images.githubusercontent.com/102961955/176442352-4b0933fd-774b-4697-8080-b0ae5727a1c2.png)

**Comment by dkazennov** [comment]
I tested the Button control in a standalone application without buttons to switch between forms (checkboxes are being used instead). It appears that if there is no buttons in the application but the tested one (closed and collected), then there will be no Button class object and Button.AccessibleObject class object in the Heap.

![image](https://user-images.githubusercontent.com/102961955/177185291-bb65770a-1544-4e0b-b395-187dd61afefe.png)

**Comment by dkazennov** [comment]
While working on the Issue I was unable to repro the memory leak in ToolStrip:
![image](https://user-images.githubusercontent.com/102961955/177513151-9c7a123e-b73b-4670-9447-04d18569acb6.png)

**Comment by dkazennov** [comment]
Retested controls - found no leaks:

- ToolStrip
- ListView
- RichTextBox
- MaskedTextBox

**Comment by dmitrii-drobotov** [comment]
I found that ListView does have a leak, if testing with Narrator or not refreshing tree in Inspect. Will be tracked in #7385.
![image](https://user-images.githubusercontent.com/102954094/177814129-3f4cb1b7-2687-4c85-94f4-a740c7818ad9.png)

**Comment by Tanya-Solyanik** [comment]
This issue was addressed in https://github.com/dotnet/winforms/issues/9224
